File: user-service/Application/DTOs/User/ReadUserDto.cs
namespace Application.DTOs.User;

public class ReadUserDto
{
    public required Guid Id { get; set; }
    public required string AvatarUrl { get; set; }
    public required string Username { get; set; }
    public required string Email { get; set; }
    public required string FirstName { get; set; }
    public required string LastName { get; set; }
    public required DateTime DateOfBirth { get; set; }
    public required DateTime CreatedAt { get; set; }
    // Relationship to current (requesting) user, if available in context
    public RelationshipStatusDto RelationshipStatus { get; set; } = RelationshipStatusDto.None;
    public Guid? FriendshipId { get; set; }
    public Guid? PrivateChatId { get; set; }
}

File: frontend/src/lib/dto/ReadUserDto.ts

export class ReadUserDto{
    id: string;
    avatarUrl: string;
    username: string;
    email: string;
    firstName: string;
    lastName: string;
    dateOfBirth: Date;
    createdAt: Date;
    relationshipStatus?: number;
    friendshipId?: string;
    privateChatId?: string;

    constructor(
        id: string,
        avatarUrl: string,
        username: string,
        email: string,
        firstName: string,
        lastName: string,
        dateOfBirth: Date,
        createdAt: Date,
        relationshipStatus?: number,
        friendshipId?: string,
        privateChatId?: string
    ) {
        this.id = id;
        this.avatarUrl = avatarUrl;
        this.username = username;
        this.email = email;
        this.firstName = firstName;
        this.lastName = lastName;
        this.dateOfBirth = dateOfBirth;
        this.createdAt = createdAt;
        this.relationshipStatus = relationshipStatus;
        this.friendshipId = friendshipId;
        this.privateChatId = privateChatId;
    }
}